backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoice/detail/InvoiceCandidateWithDetailsRepository.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * #%L
+ * de.metas.swat.base
+ * %%
+ * Copyright (C) 2024 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.invoice.detail;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import de.metas.common.util.CoalesceUtil;
+import de.metas.invoice.InvoiceLineId;
+import de.metas.invoicecandidate.InvoiceCandidateId;
+import de.metas.organization.IOrgDAO;
+import de.metas.organization.OrgId;
+import de.metas.quantity.Quantity;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.model.I_C_Invoice_Detail;
+import org.compiere.util.TimeUtil;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.Nullable;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.adempiere.model.InterfaceWrapperHelper.saveAll;
+
+@Repository
+public class InvoiceCandidateWithDetailsRepository
```

Comment:
IMHO this is a duplicate of InvoiceWithDetailsRepository.
TBH, I would prefer to
* move InvoiceCandidateId to de.metas.business
* improve existing model
* have only one repo for the same thing

Reply:
It's sort of the same, but at the same time, it isn't. InvoiceWithDetailsRepository is invoice centric, all of the operations in that class are based on the assumption that details are added to an already existing invoice/ invoice line. While in our case, we start with the details being linked to Invoice candidates, then later transferred to invoices when those are created.
Totally see what you're saying, but I'm doubtful that, in this timeframe, I can refactor as you suggested and regression test the existing invoice to details flows.